src/android/com/eclipsesource/tabris/launchmonitor/LaunchMonitor.kt

```diff
@@ -1,27 +1,40 @@
 package com.eclipsesource.tabris.launchmonitor
 
-import android.app.Activity
+import android.content.Intent
+import com.eclipsesource.tabris.android.*
 import android.net.Uri
 import android.os.Handler;
-import com.eclipsesource.tabris.android.TabrisActivity
-import com.eclipsesource.tabris.android.TabrisContext
 import java.net.URLDecoder
 import java.util.*
 
 private const val EVENT_URL_LAUNCH = "urlLaunch"
 
-class LaunchMonitor(activity: Activity, tabrisContext: TabrisContext) {
-
+class LaunchMonitor(scope: ActivityScope) {
 
   init {
-    val launchUri = (activity as TabrisActivity).intentOfCreate.getStringExtra("launchUri")
-    val queryParameters = parseQuery(launchUri)
-    val urlLaunchParameters = mapOf("queryParameters" to queryParameters, "url" to launchUri);
     Handler().post {
-      if (launchUri != null) {
-        tabrisContext.objectRegistry.getRemoteObjectForObject(this).notify(EVENT_URL_LAUNCH, urlLaunchParameters)
+      if (scope.activity.intent.dataString != null) {
+        notifyUrlLaunch(scope, scope.activity.intent)
       }
     }
+    (scope.activity.supportFragmentManager.findFragmentByTag(TabrisFragment::class.java.name) as? TabrisFragment)
+      ?.scope?.events?.addActivityStateListener(object : Events.ActivityStateListener {
```

Comment:
It should not be required to access the `TabrisFragment`.

```suggestion
      scope.events.addActivityStateListener(object : Events.ActivityStateListener {
```